includes/discovery/sensors/state/aos-emu2.inc.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * aos-emu2.inc.php
+ *
+ * LibreNMS sensors state discovery module for APC EMU2
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2018 Ben Gibbons
+ * @author     Ben Gibbons <axemann@gmail.com>
+ */
+
+ // Input Contact discovery
+
+$contacts['emu2_contacts'] = snmpwalk_cache_oid($device, 'emsInputContactStatusEntry', array(), 'PowerNet-MIB');
+
+foreach ($contacts['emu2_contacts'] as $id => $contact) {
+    $index          = $contact['emsInputContactStatusInputContactIndex'];
+    $oid            = '.1.3.6.1.4.1.318.1.1.10.3.14.1.1.3.' . $index;
+    $descr          = $contact['emsInputContactStatusInputContactName'];
+    $currentstate   = $contact['emsInputContactStatusInputContactState'];
+    $normalstate    = $contact['emsInputContactStatusInputContactNormalState'];
+    if (is_array($contacts['emu2_contacts']) && $normalstate == 'normallyClosedEMS') {
+        $state_name_nc  = 'emsInputContactNormalState_NC';
+        $state_index_id = create_state_index($state_name_nc);
+        if ($state_index_id !== null) {
+            $states = array(
+                array($state_index_id,'Normal',0,1,0) ,
```

Comment:
Most of the time it is better if you used the names returned by snmp instead of Normal and Alert. So: contactOpenEMS and contactClosedEMS.  Or at least change the description to Open and Closed ( you can keep the inverted generic values so it correctly shows up as green or red)